test/fixtures/s3-fixture/src/main/java/fixture/s3/S3HttpHandler.java

```diff
@@ -74,38 +74,35 @@ public S3HttpHandler(final String bucket) {
 
     public S3HttpHandler(final String bucket, @Nullable final String basePath) {
         this.bucket = Objects.requireNonNull(bucket);
-        this.basePrefix = Objects.requireNonNullElse(basePath, "");
-        this.path = bucket + (basePath != null && basePath.isEmpty() == false ? "/" + basePath : "");
+        this.basePath = Objects.requireNonNullElse(basePath, "");
+        this.bucketAndBasePath = bucket + (Strings.hasText(basePath) ? "/" + basePath : "");
     }
 
+    private static final Set<String> NO_REQUEST_BODY_METHODS = Set.of("GET", "HEAD", "DELETE");
```

Comment:
This name doesn't make sense. It's only a list of method strings. Maybe document how it's used instead?